yajsapi/agreement/agreement.ts

```diff
@@ -0,0 +1,80 @@
+import { Logger } from "../utils";
+import { RequestorApi } from "ya-ts-client/dist/ya-market/api";
+import { Agreement as AgreementModel, AgreementStateEnum } from "ya-ts-client/dist/ya-market/src/models";
+import { AgreementConfigContainer } from "./agreement_config_container";
+
+export { AgreementStateEnum };
+
+export interface AgreementOptions {
+  credentials?: { apiKey?: string; basePath?: string };
+  requestTimeout?: number;
+  executeTimeout?: number;
+  eventPoolingInterval?: number;
+  eventPoolingMaxEventsPerRequest?: number;
+  logger?: Logger;
+}
+
+export interface ProviderInfo {
+  providerName: string;
+  providerId: string | null;
+}
+
+export class Agreement {
```

Comment:
What if the Agreement had such an interface created by factory?
These are all the things we need right now
```js
class Agreement {
  public readonly id;
  public readonly providerId: string;
  public readonly providerName: string;

  async getState(): Promise<AgreementStateEnum> {}
  async confirm() {}
  async terminate() {}
}
```